internal/getproviders/memoize_source_test.go

```diff
@@ -121,11 +121,11 @@ func TestMemoizeSource(t *testing.T) {
 		mock := NewMockSource([]PackageMeta{meta})
 		source := NewMemoizeSource(mock)
 
-		_, err := source.AvailableVersions(nonexistProvider)
+		_, _, err := source.AvailableVersions(nonexistProvider)
 		if want, ok := err.(ErrRegistryProviderNotKnown); !ok {
 			t.Fatalf("wrong error type from first call:\ngot:  %T\nwant: %T", err, want)
 		}
-		_, err = source.AvailableVersions(nonexistProvider)
+		_, _, err = source.AvailableVersions(nonexistProvider)
```

Comment:
All of the tests in this file look like they ignore the warnings altogether, but it's important that the memoize source passes them through from its upstream source. Is there any reasonable way of adding test coverage for this? If it's too awkward then I'd be fine with leaving it, but a test would be nice if it's easy to add.